man/sd_bus_creds_get_pid.xml

```diff
@@ -394,16 +394,17 @@
 
     <para><function>sd_bus_creds_get_session()</function> will
     retrieve the identifier of the login session that the process is
-    a part of. See
-    <citerefentry><refentrytitle>systemd-logind.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>. For
-    processes that are not part of a session, returns -ENXIO.
+    a part of. Please note this function is not generally useful; see
```

Comment:
hmm, i disagree with this strong wording. the functions do have a purpose still, even though they are not as useful as they used too. Specifically, on terminal logins they do the exact same thing as they always did: group the processes directly attached to that tty session...

But also on graphical logins sessions still play a role, as they encapsulate the process that defines the lifecycle of a specific session. Or to say this differently: there's always at least one process left in the actual login session, and it is responsible for spawning the actualy display server and stuff, and stays around until the session is supposed to end. It is as such the "reference count" to the user@.service, and is defined in terms of there being a process around that hasn't exited.

I think adding some wording here along the lines of "this is not what you are looking for these days, as graphical login sessions tend to consist of stub processes only, and the main desktop environment is spawned off a shared per-user service rather than processes directly associated to the session concept"... or something like that.

Reply:
Thanks for the feedback.  I can work on everything else, but I don't understand how the wording for the docs can be weakened.

I noticed the process holding the reference count FD in footnote 1 &lt;small>in the last commit message&lt;/small>.  That's the only meaning attached to the session scope any more though.  It's not a 100% circular definition, because terminating that process (directly or by its child) will also see the terminal hung up for terminal logins.  But the ability to terminate that process is the only semantic.

You go from the session scope to the process to kill.  There's no use-case for going the opposite direction.  I don't want the docs to be raising that question if it's not actually answerable.  Neither of you gave such a counter-example?

What can you gain from a query that depends on whether or not process P was left in a session scope?  Outside of `loginctl` `user-status` / `session-status`, which again go from the unit to the process and hence do not involve `*_get_session()`.

Console logins aren't special: you can `exec systemd-run --scope sh`.  Which is no less legitimate than some `gnome-shell.target`.  (The only reason I wouldn't consider that last as the height of elegance, is you probably end up leaving a funny-looking stub process as a child of the PAM process.)

Reply:
EDIT: What you seem to be suggesting is that it could be useful to tabulate processes and include a SESSION_SCOPE column.  In that case you'd also want a mark somewhere on the processes which belonged to a user manager, and docs which explained the confusion... it seems a bit contrived...

@poettering, your comment above is on sd_bus_creds_get_pid(), not on the full description in sd_pid_get_session(), which I pointed the reader towards.  I can't guess the exact scope of your criticism :).  Also the closest to a use case I could think of from your suggestion would mainly apply to sd_pid_get_session(), not sd_bus_creds_get_pid().

I'm pretty happy if you just think it is better to inline the full description in both places, to avoid making it look so harsh.  Can you clarify that's not what you meant, and you disagree with both instances of "not generally useful"?

Reply:
I made some specific suggestions above, see https://github.com/systemd/systemd/pull/6866#discussion_r140190365.